worlds/shapez/__init__.py

```diff
@@ -0,0 +1,289 @@
+
+    def append_shapesanity(self, name: str) -> None:
+        self.shapesanity_names.append(name)
+
+    def add_alias(self, location_name: str, alias: str):
+        self.location_id_to_alias[self.location_name_to_id[location_name]] = alias
+
+    def create_regions(self) -> None:
+        menu_region = Region("Menu", self.player, self.multiworld)
+        self.location_id_to_alias = {}
+
+        # Create list of all included locations based on player options
+        self.included_locations = {**addlevels(self.maxlevel, self.options.randomize_level_logic.current_key,
+                                               self.random_logic_phase_length),
+                                   **addupgrades(self.finaltier, self.options.randomize_upgrade_logic.current_key,
+                                                 self.category_random_logic_amounts)}
+        if self.ut_active:
+            self.shapesanity_names = self.passthrough["shapesanity"]
+            self.included_locations.update(addshapesanity_ut(self.shapesanity_names, self.add_alias))
+        else:
+            self.shapesanity_names = []
+            self.included_locations.update(addshapesanity(self.options.shapesanity_amount.value, self.random,
+                                                          self.append_shapesanity, self.add_alias))
+        if self.options.include_achievements:
+            self.included_locations.update(addachievements(bool(self.options.exclude_softlock_achievements),
+                                                           bool(self.options.exclude_long_playtime_achievements),
+                                                           bool(self.options.exclude_progression_unreasonable),
+                                                           self.maxlevel,
+                                                           self.options.randomize_upgrade_logic.current_key,
+                                                           self.category_random_logic_amounts,
+                                                           self.options.goal.current_key,
+                                                           self.included_locations, self.add_alias))
+
+        self.location_count = len(self.included_locations)
+
+        # Create regions and entrances based on included locations and player options
+        self.multiworld.regions.extend(create_shapez_regions(self.player, self.multiworld, self.included_locations,
+                                                             self.location_name_to_id,
+                                                             self.level_logic, self.upgrade_logic,
+                                                             self.options.early_balancer_tunnel_and_trash.current_key,
+                                                             self.options.goal.current_key, menu_region))
+
+        # Connect Menu to rest of regions
+        main_region = self.multiworld.get_region("Main", self.player)
```

Comment:
```suggestion
        main_region = self.get_region("Main")
```
More recent shortcut helper